### DISCUSSION TITLE
Legacy library code that has contains export = syntax

### DISCUSSION THREAD

**Discussion by andrwh** [babel/babel#11649] [Q&A] [answered]
I'm working on a large codebase that has existing modules containing the following syntax:

```
const foo = {
a: 1,
}

export = foo
```

In the following link: https://babeljs.io/docs/en/babel-plugin-transform-typescript#caveats, there explicitly says there is no workaround for the `export = foo` syntax.  I was wondering if there is a way to work around this.  Unfortunately I cannot change the existing code that has these exports.  Could I use a transform? Would I have to write one?

Thanks in advance

**Comment by nicolo-ribaudo** [comment, marked as answer]
Yes, you have to write a custom transform.

Maybe you could try copy-pasting https://github.com/facebook/jest/blob/a0834f83ac7c0841b2d5782f2f8f9b2f6672e324/scripts/babel-plugin-jest-replace-ts-export-assignment.js and see if it works for you.

**Reply by G-Rath** [reply]
fwiw I've actually broken that out into its [own package/plugin](https://www.npmjs.com/package/babel-plugin-replace-ts-export-assignment), and made [a PR to mention it in the docs](https://github.com/babel/website/pull/2275) which got merged a few days ago :)
